File: 0977_Squares of a Sorted Array/SquaresofaSortedArray.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        var n = nums.Length;
        var ans = new int[n];
        var s = 0;
        var e = n - 1;
        for(int k=n-1;k>=0;k--){
            if(Math.Abs(nums[s]) > Math.Abs(nums[e]))
            {
                ans[k] = nums[s] * nums[s];
                s++;
            }else{
                ans[k] = nums[e] * nums[e];
                e--;
            }
        }
        
        return ans;
    }
}

File: 0977_Squares of a Sorted Array/SquaresofaSortedArray.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        
        int l = 0, r = n-1; 
        for(int i=n-1;i>=0;i--)
        {
            if(nums[l] * nums[l] >= nums[r] * nums[r]) {
                ans[i] = nums[l] * nums[l];
                l++;
            }else {
                ans[i] = nums[r] * nums[r];
                r--;
            }
        }
        
        return ans;
    }
}